user-api/base/deploymentconfig.yaml

```diff
@@ -177,3 +177,16 @@ spec:
             periodSeconds: 60
             timeoutSeconds: 10
   test: false
+---
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: user-api
+spec:
+  maxReplicas: 2
+  minReplicas: 1
+  scaleTargetRef:
+    apiVersion: apps.openshift.io/v1
+    kind: DeploymentConfig
```

Comment:
This can be one of the drawbacks of using `Deployment` instead of `DeploymentConfig` for Argo CD. `HorizontalPodAutoscaler` works just for `DeploymentConfig` and `ReplicationController` [1].

[1] https://docs.openshift.com/container-platform/4.5/nodes/pods/nodes-pods-autoscaling.html

Reply:
as we have a DC, let's test the HPA... @harshad16 wdyt?

Reply:
@harshad16 btw, what about using webhook triggers for triggering rollouts of DeploymentConfig from argo cd? Would that work?

Reply:
maybe @HumairAK has an updated view and some insights on the support of DeploymentConfigs by ArgoCD too?

Reply:
If not directly from argocd maybe we could do it from the build pipeline (a specific "step" for dc)?

Reply:
@goern , yes lets test Horizontal pod autoscaler.

@fridex , we can trigger the deploymentconfig through a trigger, would check it, if argocd can do this, as it would something for all. or else we can work having this triggered from pipeline.